.github/workflows/workflow.yml

```
@@ -35,53 +39,27 @@ jobs:
         uses: ocaml/setup-ocaml@v2
         with:
           ocaml-compiler: ${{ matrix.ocaml-compiler }}
-          dune-cache: true
+          dune-cache: ${{ matrix.os == 'ubuntu-latest' }}
           opam-depext: true
           opam-depext-flags: --with-test
+          opam-local-packages: ${{ matrix.local-packages }}
 
       - run: yarn --frozen-lockfile
         working-directory: cohttp-lwt-jsoo/test
 
-      - run: opam depext -i re uri uri-sexp sexplib0 ppx_sexp_conv stringext fmt alcotest conf-libev lwt cmdliner conduit-lwt-unix magic-mime logs ounit js_of_ocaml-lwt js_of_ocaml-ppx async async_unix base core core_unix conduit-async mirage-crypto ipaddr conduit-mirage mirage-kv astring base_quickcheck ppx_assert ppx_compare ppx_here core_bench crowbar ocurl mirage-channel lwt_ppx core_kernel.v0.14.2
-
-      - run: opam exec -- dune build -p http,cohttp,cohttp-lwt,cohttp-lwt-unix,cohttp-server-lwt-unix,cohttp-lwt-jsoo,cohttp-mirage,cohttp-async,cohttp-curl-async,cohttp-curl-lwt,cohttp-curl,cohttp-top --profile release
-
-      - run: opam exec -- dune runtest cohttp cohttp-async cohttp-curl cohttp-curl-async cohttp-curl-lwt cohttp-lwt cohttp-lwt-unix cohttp-mirage cohttp-server-lwt-unix cohttp-top cohttp_server http
-
-      - run: opam exec -- make js-test
-
-  build-test-macos:
-    strategy:
-      fail-fast: false
-      matrix:
-        os:
-          - macos-latest
-        ocaml-compiler:
-          - 4.14.x
-
-    runs-on: ${{ matrix.os }}
-
-    steps:
-      - name: Checkout code
-        uses: actions/checkout@v3
-
-      - name: Use OCaml ${{ matrix.ocaml-compiler }}
-        uses: ocaml/setup-ocaml@v2
-        with:
-          ocaml-compiler: ${{ matrix.ocaml-compiler }}
-          dune-cache: false
-          opam-depext: true
-          opam-depext-flags: --with-test
-
       - run: echo "PKG_CONFIG_PATH=$(brew --prefix openssl)/lib/pkgconfig" >>"$GITHUB_ENV"
+        if: ${{ matrix.os == 'macos-latest' }}
 
       - run: opam install --with-test --deps-only http cohttp cohttp-lwt cohttp-lwt-unix cohttp-server-lwt-unix cohttp-mirage cohttp-async cohttp-curl-async cohttp-curl-lwt cohttp-curl cohttp-top
 
       - run: |
           opam exec -- dune build --profile=dev --only-packages http,cohttp,cohttp-lwt,cohttp-lwt-unix,cohttp-server-lwt-unix,cohttp-mirage,cohttp-async,cohttp-curl-async,cohttp-curl-lwt,cohttp-curl,cohttp-top @install
-          opam exec -- dune build -j1 bench/ examples/ cohttp-curl-async/bin/ cohttp-curl-lwt/bin/
+          opam exec -- dune build bench/ examples/ cohttp-curl-async/bin/ cohttp-curl-lwt/bin/
+      
       - run: opam exec -- dune runtest --profile=dev cohttp cohttp-async cohttp-curl cohttp-curl-async cohttp-curl-lwt cohttp-lwt cohttp-lwt-unix cohttp-mirage cohttp-server-lwt-unix cohttp-top http
 
+      - run: opam exec -- make js-test
+
```

You need
```suggestion
      - if: ${{ matrix.os == 'ubuntu-latest' }} 
        run: |
          opam depext -i cohttp-lw-jsoo
          opam exec -- make js-test
```
Or something like it

---

add cohttp-lwt-jsoo to `opam  install`